markdown/extensions/toc.py

```diff
@@ -347,6 +353,10 @@ def __init__(self, **kwargs):
             "permalink_title": ["Permanent link",
                                 "Title attribute of the permalink - "
                                 "Defaults to 'Permanent link'"],
+            "permalink_leading": [0,
```

Comment:
I realize that the `permalinks` option uses `0` as a default for `False`, but that is a very old option which existed prior to any option validation. Instead, let's use `False` as the default value as we do with the `anchorlink` option. Sorry that I didn't notice this before.

We should probably change the `permalink` default at some point, but that would be separate from this PR.

Reply:
No problem, easy enough to change.